contextid.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"context"
+	"net/http"
+)
+
+// AddContextID wraps a http.Handler to add a monotonically increasing
+// uint to the context of the http.Request with the key "id" (string)
+// as it passes through the request to the next handler.
+func AddContextID(next http.Handler) http.Handler {
+	ids := make(chan uint)
+	go func() {
+		for id := uint(0); ; id++ {
+			ids <- id
+		}
+	}()
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx := context.WithValue(req.Context(), "id", <-ids)
```

Comment:
I think it's recommended not to use primitive types as context keys. If some other middleware adds a value with key "id" your id gets overridden.
`context.WithValue(parentCtx, key, value)` takes `key` parameter of type `interface{}`.
Two interface values are equal if they have identical dynamic types and equal dynamic values or if both have value nil.
See https://golang.org/pkg/context/#WithValue and https://golang.org/ref/spec#Comparison_operators

Reply:
Yes, that's correct. You could create an issue for it. But this is how the code was already - I've just moved where the id is generated and added. I don't plan to fix that (in this PR, or maybe at all).